storage/netapp/netapp_e_hostgroup.py

```diff
@@ -0,0 +1,426 @@
+                group_name,
+                ssid,
+                str(err)))
+
+    hosts_in_group = []
+
+    for hg in g_data:
+        if hg['name'] == group_name:
+            clusterRef = hg['clusterRef']
+
+    for host in h_data:
+        if host['clusterRef'] == clusterRef:
+            hosts_in_group.append(host['name'])
+
+    return hosts_in_group
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            name=dict(required=False),
+            new_name=dict(required=False),
+            ssid=dict(required=True),
+            id=dict(required=False),
+            state=dict(required=True, choices=['present', 'absent']),
+            hosts=dict(required=False, type='list'),
+            api_url=dict(required=False),
+            api_username=dict(required=False),
+            validate_certs=dict(required=False, default=True),
+            api_password=dict(required=False, no_log=True)
+        ),
+        supports_check_mode=False,
+        mutually_exclusive=[['name', 'id']],
+        required_one_of=[['name', 'id']]
+    )
+
+    name = module.params['name']
+    new_name = module.params['new_name']
+    ssid = module.params['ssid']
+    id_num = module.params['id']
+    state = module.params['state']
+    hosts = module.params['hosts']
+    user = module.params['api_username']
+    pwd = module.params['api_password']
+    api_url = module.params['api_url']
+
+    if not api_url.endswith('/'):
+        api_url += '/'
+
+    if name:
+        id_type = 'name'
+        id_key = name
+    elif id_num:
+        id_type = 'id'
+        id_key = id_num
+
+    exists, group_data = group_exists(module, id_type, id_key, ssid, api_url, user, pwd)
+
+    if state == 'present':
+        if not exists:
+            try:
+                rc, data = create_hostgroup(module, ssid, name, api_url, user, pwd, hosts)
+            except Exception:
+                err = get_exception()
+                module.fail_json(msg="Failed to create a host group. Id [%s]. " +
+                                     "Error [%s]." % (name, ssid, str(err)))
+
+            hosts = get_hosts_in_group(module, ssid, name, api_url, user, pwd)
+            module.exit_json(changed=True, hosts=hosts, **data)
+        else:
+            current_hosts = get_hosts_in_group(module, ssid, name, api_url, user, pwd)
+
+            if not current_hosts:
+                current_hosts = []
+
+            if not hosts:
+                hosts = []
+
+            if set(current_hosts) != set(hosts):
+                try:
+                    rc, data = update_hostgroup(module, ssid, name, api_url, user, pwd, hosts, new_name)
+                except Exception:
+                    err = get_exception()
+                    module.fail_json(msg="Failed to update host group. Group: [%s]. Id [%s]. " +
+                                         "Error [%s]." % (name, ssid, str(err)))
```

Comment:
MUSTFIX: E:403,41: Too many arguments for format string (too-many-format-args)

(Only the second part of the '+'ed msg ["Error [%s]"] gets the args...)